### PR TITLE
Oncoprint heatmap clustering improvements

### PR DESCRIPTION
**Pull request by pieterlukasse**
# What? Why?
Improved clustering distance function to push samples with no data to outside. This solution should push samples that have no data to the outer edge of the resulting hierarchy (and of the resulting oncoprint heatmap). See screenshots.

**BEFORE**
image

**NOW (with solution propose here)**
image

Additional improvements:
- added missing LICENSE for clustering js library
- added support for clustering (sorting) on one track only (see screenshot below)

image

### CODE REVIEW

File: portal/src/main/webapp/js/src/clustering/clustering-worker.js [hunk 1]
```diff
@@ -54,11 +68,26 @@ var preRankedSpearmanDist = function(item1, item2) {
 	//take the arrays from the preProcessedValueList:
 	var ranks1 = item1.preProcessedValueList;
 	var ranks2 = item2.preProcessedValueList;
+	var item1AllNaNs = isAllNaNs(item1.orderedValueList);
```
Comment (adamabeshouse) [inline portal/src/main/webapp/js/src/clustering/clustering-worker.js]: this should probably be precomputed

Comment (pieterlukasse) [inline portal/src/main/webapp/js/src/clustering/clustering-worker.js]: hi @adamabeshouse, good point. Will do.

**Comment by pieterlukasse** [comment]
@adamabeshouse I updated the code and also made a small fix for an existing bug (before this PR) I noticed during testing.

**Comment by adamabeshouse** [comment]
Code looks good but heroku is crashing, I think you need to rebase onto rc

**Comment by pieterlukasse** [comment]
@adamabeshouse rebased again.

**Comment by jjgao** [comment]
@pieterlukasse can we make this to `hotfix` or `release-1.6.0`?

**Comment by pieterlukasse** [comment]
@jjgao: changed to `hotfix`

**Comment by adamabeshouse** [comment]
@pieterlukasse I'm still seeing errors on Heroku and when I tried to run it locally using the database that's used for hotfix (cgds_public), it says theres a database version mismatch.

**Comment by pieterlukasse** [comment]
@adamabeshouse the database version mismatch is not related to this PR, but is rather a problem in the DB itself (it needs to be upgraded to the most recent version). Adding @sheridancbio to this thread (I think he is somehow involved in upgrading these DBs - see also #1806, recently merged)

**Pull request merged**